# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build xorg-server-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.63; am-1.11; lt-2.2.6)
libtoolize -c -f; aclocal -I m4; autoheader; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig ./configure --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/lib/xorg/modules/input
install src/.libs/evdev_drv.so $DESTDIR/x11/lib/xorg/modules/input

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/xf86-input-evdev-build.pc << EOF
Name: xf86-input-evdev
Version: 2.3.0
Description: event device input driver for the X server
Requires: $REQUIRES

exec=\\
/x11/lib/xorg/modules/input/evdev_drv.so
EOF

exit

# patch
--- xf86-input-evdev-2.3.0.orig/configure.ac
+++ xf86-input-evdev-2.3.0/configure.ac
@@ -28,6 +28,7 @@
 
 AC_CONFIG_SRCDIR([Makefile.am])
 AC_CONFIG_AUX_DIR(.)
+AC_CONFIG_MACRO_DIR([m4])
 AM_INIT_AUTOMAKE([dist-bzip2])
 
 AM_MAINTAINER_MODE
--- xf86-input-evdev-2.3.0.orig/src/evdev.c
+++ xf86-input-evdev-2.3.0/src/evdev.c
@@ -1530,10 +1530,10 @@
         EvdevInitAnyClass(device, pEvdev);
     else if (pEvdev->flags & (EVDEV_TOUCHPAD | EVDEV_TOUCHSCREEN | EVDEV_TABLET))
         EvdevInitTouchDevice(device, pEvdev);
-    else if (pEvdev->flags & EVDEV_RELATIVE_EVENTS)
-        EvdevInitRelClass(device, pEvdev);
     else if (pEvdev->flags & EVDEV_ABSOLUTE_EVENTS)
         EvdevInitAbsClass(device, pEvdev);
+    else if (pEvdev->flags & EVDEV_RELATIVE_EVENTS)
+        EvdevInitRelClass(device, pEvdev);
 
 #ifdef HAVE_PROPERTIES
     /* We drop the return value, the only time we ever want the handlers to

# source
https://x.org/releases/individual/driver/xf86-input-evdev-2.3.0.tar.gz
